iree/tools/CMakeLists.txt

```diff
@@ -416,3 +416,5 @@ if(${IREE_ENABLE_MLIR})
     )
   endif()
 endif()
+
+add_subdirectory(android)
```

Comment:
We'll need the `vm_util` target defined for the `android` directory. I think one way to address it is to put the `vm_util` under a `utils` subdirectory. I've made the change as the last commit. If that looks fine, I'm happy to either land it as a part of this pull request, or revert the last change and land it separately.

Reply:
Do we? Maybe I'm misremembering, but isn't the CMake configure step tolerant to declaring a dependency on a target that has not itself been declared yet?

Reply:
Sorry I was missing context here: the `android` directory compiles a `.so`, which is different than the normal `.a` static libraries we have been handling thus far. We have complicated always link logic and other logic handling dependencies. The *symptom* I observed is that if `utils/` was added *after* `android/` then `vm_util` is missing as a dependency to the `run_module_app.so`. If it is added *before* `android/`, then everything is fine. I haven't gotten the time to dive into understanding why this particular behavior; it can also indicate a bug somewhere. If you or @GMNGeoffrey has quick thoughts on why this behavior that would be awesome. :)

Reply:
Created https://github.com/google/iree/issues/3317 to track this.